b2/encryption.py

```diff
@@ -0,0 +1,174 @@
+######################################################################
+#
+# File: b2/encryption.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import os
+import struct
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes, hmac, padding
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+from cryptography.hazmat.primitives.ciphers import (Cipher, algorithms, modes)
+
+
+def derive_key(passphrase, salt, iterations=500000):
+    kdf = PBKDF2HMAC(
+        algorithm=hashes.SHA256(),
+        length=16,
+        salt=salt,
+        iterations=iterations,
+        backend=default_backend()
```

Comment:
I think they won't change the backend to something which supports less features. The only really useful backend at the moment seems to be OpenSSL. But I can change the code so that we can easily change the backend when required.